src/ui_fsmenu/addons/manager.cc

```diff
@@ -850,8 +851,36 @@ bool AddOnsCtrl::handle_key(bool down, SDL_Keysym code) {
 }
 
 void AddOnsCtrl::refresh_remotes() {
+	UI::ProgressWindow progress(this, "", "");
+	const std::string step_message = _("Fetching add-ons (%.1f%%)");
+
 	try {
-		remotes_ = net().refresh_remotes();
+		progress.step(_("Connecting to the server…"));
+
+		std::vector<std::string> names = net().refresh_remotes();
+
+		const int64_t nr_orig_entries = names.size();
+		int64_t nr_addons = nr_orig_entries;
+		remotes_.resize(nr_addons);
+
+		for (int64_t i = 0, counter = 0; i < nr_addons; ++i, ++counter) {
+			progress.step((boost::format(step_message) % (100.0 * counter / nr_orig_entries)).str());
+
+			try {
+				remotes_[i].reset(new AddOns::AddOnInfo(net().fetch_one_remote(names[i])));
+
+				if (!remotes_[i]->matches_widelands_version()) {
```

Comment:
By the way, I intend to add a server-side check soon to ensure that add-ons that are incompatible with the Widelands version in use are not listed by the server in the first place.